plugins/pulp_docker/plugins/migrations/0006_add_arch_os_size_fields.py

```diff
@@ -0,0 +1,58 @@
+import logging
+import os
+
+from pulp.server.db.connection import get_collection
+from pulp.server.db.migrations.lib import utils
+
+
+_logger = logging.getLogger('pulp_docjer.plugins.migrations.0006')
+
+
+def migrate(*args, **kwargs):
+    migrate_manifest_list()
+    migrate_blob()
+
+
+def migrate_manifest_list():
+    """
+    Update format of manifest on manifest lists to have digest, os, and arch fields
+    """
+
+    manifest_list_collection = get_collection('units_docker_manifest_list')
+    total_manifest_lists = manifest_list_collection.count(
+        {'manifests': {'$type': ["string"]}})
+
+    with utils.MigrationProgressLog('Update manifest field on ManifestList',
+                                    total_manifest_lists) as migration_log:
+        for manifest_list in manifest_list_collection.find(
+                {'manifests': {'$type': ["string"]}}).batch_size(100):
+
+            manifests = []
+            for digest in manifest_list['manifests']:
+
+                manifest = {'digest': digest,
+                            'os': '',
```

Comment:
i think the initial thought was to populate these fields based on the parsed json file which can be found at _storage_path location.
if we leave it like this, these fields will be empty forever, even after re-sync.

Reply:
Ah I did not realize the information was available in the json file on disk. I checked the database and couldn't find this data stored... I'll update this to read and obtain the data from disk

Reply:
```> db.units_docker_manifest_list.find()[0]
{
	"_id" : "265ffcf6-eadc-4ded-94f5-a9cb4b33c3ee",
	"pulp_user_metadata" : {
		
	},
	"_last_updated" : 1529492506,
	"_storage_path" : "/var/lib/pulp/content/units/docker_manifest_list/ef/19054b2d80f7ccdf1c9b8fdd78843b1671948ea691f46e31a457eabfc74849/sha256:a25f6b01e0889ce3bdaf8cd16d1d761bc8ec094af9ee78d5937d06477a5646db",
	"downloaded" : true,
	"digest" : "sha256:a25f6b01e0889ce3bdaf8cd16d1d761bc8ec094af9ee78d5937d06477a5646db",
	"schema_version" : 2,
	"manifests" : [
		{
			"arch" : "",
			"os" : "",
			"digest" : "sha256:4377b7122aed8ab432dd76be23577ffaa479febd432e4243c10946ae4d8f0ce8"
		}
	],
	"amd64_digest" : "sha256:4377b7122aed8ab432dd76be23577ffaa479febd432e4243c10946ae4d8f0ce8",
	"amd64_schema_version" : 2,
	"_ns" : "units_docker_manifest_list",
	"_content_type_id" : "docker_manifest_list"
}
[ipanova@ina pulp_docker]$ cat /var/lib/pulp/content/units/docker_manifest_list/ef/19054b2d80f7ccdf1c9b8fdd78843b1671948ea691f46e31a457eabfc74849/sha256:a25f6b01e0889ce3bdaf8cd16d1d761bc8ec094af9ee78d5937d06477a5646db
{
    "manifests": [
        {
            "digest": "sha256:4377b7122aed8ab432dd76be23577ffaa479febd432e4243c10946ae4d8f0ce8",
            "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
            "platform": {
                "architecture": "amd64",
                "os": "linux"
            },
            "size": 949
        }
    ],
    "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
    "schemaVersion": 2
}[ipanova@ina pulp_docker]$